CONTRIBUTING.md

```diff
@@ -175,3 +175,90 @@ For testing and development you can find self-signed certificates under
+users errors can be for example, a stream could not be created due to incorrect user data.
+The following questions should be answered with these log messages.
+- when did the error occur
+- what data was used to cause the error
+- what api path did the user call
+- ...
+
+#### Exclude sensitive information
+Be careful when working on parts of the application that could internationally contain sensitive data.
+In these cases, it could be useful to not include the data payload when logging.
+In some cases, an object id could be provided for more context, such as a Tenant/UserID instead
+of a username but only where the id belongs to a persisted object that an administrator can access.
+
+#### Avoid adding too much Context
+Having too much information in a log message is a thing. Image a deploymentID being logged for
+an object before it has been saved to the database. The error causes it to not be persisted, but the
+admin assumes it should be in the DB. This would waste time in searching for the object
+and potentially cause the assumption of an error that doesn't actually exist.
+
+### When information should be logged.
+Information doesn't need to be logged everywhere in the application.
+```
+{
+  "level": "info",
+  "message": "User logged in succesfully",
+  "timestamp": "2021-11-03T20:44:08.460Z"
+}
+```
+just doesn't need to pop up in the application logs.
+Information SHOULD be logged when it could potentially provide extra context when an error occurs.
+
+Some examples of when to log information:
+- An unexpected exception has occurred (e.g. a timeout exception)
+  - Provide more context: Where did the timeout occur? What data payload caused the timeout?
+- Authentication Exception (don't log an unauthorized exception, don't log because of incorrect credentials)
+  - Provide more context: What provider was used?
+- Too many failed attempts
+  - You have a process that sometimes fails because a service hasn't started yet so your code
+     retries periodically. After `X`-amount of attempt a message could be logged, since the service
+     might actually be unavailable.
+
+some example of when to NOT log information:
+- An entity could not be updated when rest endpoint is called
+  - this information should be returned to the user
+- An object was created successfully
+- A standard object has been created (such as a pipeline object)
+- A standard application process has begun (such as `starting validation process`)
+
+### How information should be logged.
+#### Log in JSON Format
+We should aim to always log messages in JSON format. JSON format is not only easily user readable
+it also allows for seamless integrations with external log management tools for extra flexibility
+and a higher observability.
+The logging framework we use at DataCater automatically appends the logger messages into JSON format.
+
+
+#### Use the correct logging level:
```

Comment:
This great. Maybe we can enhance with examples, like 
``` java
try {...} 
catch (SomeRuntimeException sre) { 
  LOGGER.debug("Exception occured, although I am a genius developer.");
  throw sre;
}
```

Maybe give a small example for each level. For info this might somehting like:

```java
public void startup(@Observes StartupEvent event) {
  LOGGER.info("Loading xyz during startup. Stay tuned ...");
}
```